src/app/data-storage.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { tap } from 'rxjs/operators';
+
+import { OwnerService } from './owner/owner.service';
+import { Owner } from './owner/owner.model';
+import { PetService } from './pets/pet.service';
+import { Pet } from './pets/pet.model';
+
+@Injectable({ providedIn: 'root' })
+export class DataStorageService {
+    constructor(private http: HttpClient, private ownerService: OwnerService, private petService: PetService) { }
+
+    storeOwner() {
+        const owner = this.ownerService.getOwners()
+        this.http
+            .put(
+                'http://localhost:8080/owner',
```

Comment:
http://localhost:8080 
this string is being used many many times, should move it to a constant and reuse it
Also move this as a baseUrl to environment.ts, so incase it needs to be changes, the code doesnt need to be compiled again